syntax = "proto3";

import "google/protobuf/empty.proto";

package course;

message Course {
    int32 id = 1;
    string title = 2;
    int32 instructor_id = 3;
}

message IdRequest {
    int32 id = 1;
}

message BaseResponse {
    bool success = 1;
    string message = 2;
}

message GetCoursesByInstructorRequest {
    int32 instructor_id = 1;
}

message GetAllCoursesResponse {
    repeated Course courses = 1;
}

service CourseService {
    rpc GetCourseById(IdRequest) returns (Course);
    rpc GetCoursesByInstructor(GetCoursesByInstructorRequest) returns (GetAllCoursesResponse);
    rpc GetAllCourses(google.protobuf.Empty) returns (GetAllCoursesResponse);
    rpc CreateCourse(Course) returns (BaseResponse);
    rpc UpdateCourse(Course) returns (BaseResponse);
    rpc DeleteCourse(IdRequest) returns (BaseResponse);
}
